src/core/utils/props.ts

```diff
@@ -48,3 +48,10 @@ export type PolymorphicComponentPropsWithRef<
 > = PolymorphicComponentProps<T, Props> & {
   ref?: React.ComponentPropsWithRef<T>['ref'];
 };
+
+export type AsProp<T> = {
+  /**
+   * What element should this component be rendered as?
+   */
+  as?: T;
```

Comment:
What was the issue that you need this type?

Reply:
we are extending `ButtonProps` in a few other components, where it would not work without adding logic